platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/permission/PermissionManager.java

```diff
@@ -10,57 +10,115 @@
  *******************************************************************************/
 package org.eclipse.che.api.core.rest.permission;
 
-import com.google.inject.Inject;
-import org.eclipse.che.api.core.ApiException;
-import org.eclipse.che.api.core.BadRequestException;
 import org.eclipse.che.api.core.ForbiddenException;
 import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.commons.annotation.Nullable;
 
-import javax.inject.Singleton;
-import javax.ws.rs.core.SecurityContext;
-import java.util.HashMap;
+import javax.validation.constraints.NotNull;
+import java.util.Collections;
 import java.util.Map;
-import java.util.Set;
 
 /**
+ * Manages access to named operations.
  *
- * @author gazarenkov
+ * @author Eugene Voevodin
  */
-@Singleton
-public class PermissionManager {
+public interface PermissionManager {
 
-    private final Map<String, PermissionChecker> pcheckers;
+    /**
+     * Checks that user with given id has access to perform the operation.
+     *
+     * @param operation
+     *         operation for which access should be allowed or rejected
+     * @param params
```

Comment:
Is it possible that some parameter is mandatory for an operation. Which exception will be thrown in that case?